### ISSUE TITLE
CR not supported within quoted text fields in CSV Source file

### ISSUE THREAD

**Issue by kbraak** [gbif/ipt#1135]
```
What steps will reproduce the problem?
1. Create a resource
2. Add CSV Source data : UTF-8, delimiter=;, quote="
3. Choose a file with multiline fields : <CR> within quoted fields.

IPT report problems when importing the CSV file:
Message mention "Missing columns" in CSV source file.


What version of the provider software are you using? The version should be
displayed in the footer of any page. 2.1.1-r4640 


What browser are you using? Chrome Version 39.0.2171.95 (64-bit)


Please provide any additional information below.
Workaround: remove <CR><LF> form quoted fields.

```

Original issue reported on code.google.com by `AndreJJH` on 2015-01-13 09:30:42

**Comment by krkabol** [comment]
This is still a rather surprising limitation for a tool that accepts CSV and is widely used for publishing Darwin Core data.

A quoted CSV field is explicitly allowed to contain line breaks. For example, this is valid CSV:

```csv
occurrenceID,locality
123,"Some locality
with a second line"
```

The same applies to carriage returns within a quoted field. A CSV parser should treat the quoted field as one field/record rather than interpreting the embedded line break as the end of the record.

Darwin Core does not impose a restriction that textual values must not contain CR or LF either. Darwin Core defines the terms and their semantics; the serialization layer is responsible for representing those values correctly.

**The most concerning aspect of this limitation is that it does not necessarily result in a clear import error.** Depending on where the line break occurs, the result can be much harder to detect: individual records may disappear, columns may become shifted, or records may be split and combined in unexpected ways. This can even produce seemingly valid records with nonsensical identifiers, or misleading validation errors such as duplicate `occurrenceID` values.

In other words, the problem is not simply that an import fails. It can potentially result in silently corrupted or incomplete published data, which is much more difficult to detect and diagnose.

I understand that the current workaround is to remove CR/LF characters before importing the data, but that effectively imposes a restriction on otherwise valid CSV/Darwin Core data.

It would be much more robust if IPT used a CSV parser capable of correctly handling quoted fields containing CR/LF (and replacing it, if neccessary).

Could this limitation be reconsidered?

**Comment by mike-podolskiy90** [comment]
Thank you, @krkabol 

You can find it in the manual that we advise to remove line break characters https://ipt.gbif.org/manual/en/ipt/latest/data-preparation#text-files-source. The help icon to `Field quotes` also advises that:

<img width="672" height="315" alt="Image" src="https://github.com/user-attachments/assets/f901ee03-15b4-4cd8-8e0d-6e01a602ffe4" />

We can probably do something to display a more visible error/warning to make it more robust.

**Comment by krkabol** [comment]
Thank you. I am aware of the documented workaround. My point, however, was to argue for changing the IPT's behaviour, rather than documenting a workaround for data that is otherwise valid CSV.

A CSV parser should handle CR/LF inside quoted fields correctly. Requiring users to remove them effectively imposes an unnecessary restriction and, much more importantly, can lead to silent data corruption rather than a clear import error.

If changing the parser is not feasible or would have important implications for the IPT, a simple check for CR/LF characters before parsing, resulting in a clear import error, would already be a significant improvement.

**Comment by mike-podolskiy90** [comment]
> If changing the parser is not feasible or would have important implications for the IPT, a simple check for CR/LF characters before parsing, resulting in a clear import error, would already be a significant improvement.

Yes, I am thinking IPT should remove/replace those characters and display a clear warning that they were removed/replaced.

**Comment by mike-podolskiy90** [comment]
https://github.com/gbif/ipt/issues/3187
